l10n_th_withholding_tax_cert/tests/test_wt_cert.py

```diff
@@ -156,26 +158,108 @@ def test_01_create_wt_cert_payment(self):
         wizard = f.save()
-        wizard.write({"wt_account_ids": invoice_id.line_ids.mapped("account_id")})
+        wizard.write({"wt_account_ids": invoice.line_ids.mapped("account_id")})
         res = wizard.create_wt_cert()
         # New WT Cert
         ctx_cert = res.get("context")
-        ctx_cert.update({"default_income_tax_form": "pnd3", "wt_cert_income_type": "1"})
+        ctx_cert.update({"income_tax_form": "pnd3", "wt_cert_income_type": "1"})
         with Form(self.wt_cert.with_context(ctx_cert)) as f:
             f.income_tax_form = "pnd3"
         wt_cert = f.save()
         self.assertEqual(wt_cert.supplier_partner_id, self.partner_1)
-        invoice_id.button_wt_certs()
+        invoice.button_wt_certs()
+
+    def test_03_create_wt_cert_payment_multi(self):
+        """ Payments to WT Certs """
+        bank_journal = self.browse_ref("account.bank_journal")
+        expenses_journal = self.browse_ref("account.expenses_journal")
+        invoice = self._create_invoice(
+            self.partner_1.id, expenses_journal.id, "in_invoice"
+        )
+        invoice.action_post()
+        # Payment by writeoff with withholding tax account
+        ctx = {
+            "active_ids": [invoice.id],
+            "active_id": invoice.id,
+            "active_model": "account.move",
+        }
+        view_id = "account.view_account_payment_invoice_form"
+        with Form(self.account_payment.with_context(ctx), view=view_id) as f:
+            f.journal_id = bank_journal
+            f.amount = 97.0  # To withhold 3.0
+            f.payment_difference_handling = "reconcile"
+            f.writeoff_account_id = self.wt_account
+            f.writeoff_label = "Withhold 3%"
+        payment = f.save()
+        payment.post()
+        self.assertEqual(payment.state, "posted")
+        payment2 = payment.copy()
+        payment2.post()
+        self.assertEqual(payment2.state, "posted")
+        # Create WT Cert from Payment's Action Wizard
+        ctx = {
+            "active_ids": [payment.id, payment2.id],
+            "active_model": "account.payment",
+        }
+        res = self.wt_cert.with_context(ctx).action_create_withholding_tax_cert()
+        view = self.env["ir.ui.view"].browse(res["view_id"]).xml_id
+        with Form(
+            self.env[res["res_model"]].with_context(res["context"]), view=view
+        ) as f:
+            f.income_tax_form = "pnd3"
+            f.wt_cert_income_type = "1"
+        wizard = f.save()
+        wizard.write({"wt_account_ids": invoice.line_ids.mapped("account_id")})
+        res = wizard.create_wt_cert_multi()
+        certs = self.wt_cert.search(res["domain"])
+        self.assertEqual(len(certs), 2)
+
+    def test_04_create_wt_cert_je_multi(self):
+        """ Journal Entries to WT Certs """
+        misc_journal = self.browse_ref("account.miscellaneous_journal")
+        invoice = self._create_invoice(False, misc_journal.id, "entry", self.wt_account)
```

Comment:
better name it "journal_entry" or "move", not "invoice" (this is not invoice, isn't it?)